logstash-core/src/main/java/org/logstash/execution/QueueBatch.java

```diff
@@ -0,0 +1,13 @@
+package org.logstash.execution;
+
+import org.jruby.RubyArray;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.IOException;
+
+public interface QueueBatch {
+    int filteredSize();
+    RubyArray toRubyArray();
+    void merge(IRubyObject event);
```

Comment:
@danhermann shouldn't we be able to do `JRubyEventExt.RubyEvent` here for the type?

Reply:
Per our conversation, we'll do this in a later PR.